pkg/vsphere/disk/util.go

```diff
@@ -242,15 +239,49 @@ func findDisk(op trace.Operation, vm *vm.VirtualMachine, name string) (*types.Vi
 			return false
 		}
 
-		op.Debugf("backing file name %s", backing.VirtualDeviceFileBackingInfo.FileName)
-		match := strings.HasSuffix(backing.VirtualDeviceFileBackingInfo.FileName, name)
-		if match {
-			op.Debugf("Found candidate disk for %s at %s", name, backing.VirtualDeviceFileBackingInfo.FileName)
+		backingFileName := backing.VirtualDeviceFileBackingInfo.FileName
+		op.Debugf("backing file name %s", backingFileName)
+
+		return filter(backingFileName)
+	})
+
+	if len(candidates) == 0 {
+		return nil, nil
+	}
+
+	disks := make([]*types.VirtualDisk, len(candidates))
+	for idx, disk := range candidates {
+		d, ok := disk.(*types.VirtualDisk)
+		if !ok {
+			// will never happen
```

Comment:
if it never happens, this logic should not be here. You can just say the pre-filtering already happened right before this.